aws/tagsECR_test.go

```diff
@@ -0,0 +1,79 @@
+package aws
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecr"
+)
+
+// go test -v -run="TestDiffECRTags"
+func TestDiffECRTags(t *testing.T) {
```

Comment:
I think we should check that we aren't submitting overlapping tags, see also:

https://github.com/terraform-providers/terraform-provider-aws/blob/d24f4f4989f33b0255c9d5066b0b1be891e577a3/aws/tags_test.go#L50-L66

The fix is:

https://github.com/terraform-providers/terraform-provider-aws/blob/d24f4f4989f33b0255c9d5066b0b1be891e577a3/aws/tags.go#L202-L205